rust_src/src/hashtable.rs

```diff
@@ -207,12 +221,9 @@ pub fn copy_hash_table(mut table: LispHashTableRef) -> LispHashTableRef {
 /// If KEY is not found, return DFLT which defaults to nil.
 #[lisp_fn(min = "2")]
 pub fn gethash(key: LispObject, hash_table: LispHashTableRef, dflt: LispObject) -> LispObject {
-    let idx = hash_table.lookup(key, ptr::null_mut());
-
-    if idx >= 0 {
-        hash_table.get_hash_value(idx)
-    } else {
-        dflt
+    match hash_table.lookup(key) {
+        Found(idx) => hash_table.get_hash_value(idx),
+        Missing(_) => dflt,
     }
```

Comment:
I love this

Reply:
Yeah, I pointed @brotzeit at it and then realized how poor the UX was.

Reply:
I suppose we could get fancy and add `and_then` or `or_else` to it.